### PR TITLE
feat(plugin-essentials): add support for glob patterns to yarn remove and yarn up

### LINKED ISSUE #1418: [Feature] Support glob patterns with yarn up
- [ ] I'd be willing to implement this feature
- [ ] This feature can already be implemented through a plugin

**Describe the user story**

As already described in [#1414](https://github.com/yarnpkg/berry/issues/1414) it would be great if there were a way to upgrade multiple dependencies - e.g. all dependencies with a specific scope.

**Describe the solution you'd like**

As @arcanis already mentioned, it would be great if the `yarn up` command would support glob patterns.
> I don't think `yarn up` supports glob patterns yet, but it seems a good idea and we'd likely merge a PR to this effect ! Then you'd just run yarn up `@company/*@canary`.

### PR DESCRIPTION
**Pull request by paul-soporan**
**What's the problem this PR addresses?**

`yarn remove` and `yarn up` don't currently support glob patterns, which makes removing / upgrading all dependencies containing a specific scope / word more difficult than it should be.

Closes #1418.

**How did you fix it?**

I implemented glob support to both `yarn remove` and `yarn up`, using `micromatch`, the library that we already use for everything glob related.

For `yarn remove`, we get the list of idents for a scope and we iterate over the matches, removing the descriptor corresponding to the matched ident. 

For `yarn up`, we get the list of idents for a scope and we iterate over the matches, setting the existing descriptor to the descriptor corresponding to the matched ident, and the descriptor used in the request to a new descriptor, created from the matched ident and the range of the pattern. For obvious reasons, the range has to be static, so glob patterns can only be used for the ident part of the descriptor.

I haven't added any tests yet, but locally all of the examples I added to the documentation work, so nothing that previously worked should be broken.

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I have verified that all automated PR checks pass.

### CODE REVIEW

File: packages/plugin-essentials/sources/commands/remove.ts [hunk 1]
```diff
@@ -89,11 +89,16 @@ export default class RemoveCommand extends BaseCommand {
 
         for (const target of targets) {
           const descriptors = workspace.manifest.getForScope(target);
-          const stringifiedIdents = [...descriptors.values()].map(structUtils.stringifyIdent);
+          const stringifiedIdents = [...descriptors.values()].map(stringifiedIdent => {
```
Comment (paul-soporan) [inline packages/plugin-essentials/sources/commands/remove.ts:92]: Technically, they're not `stringifiedIdent`s, but `descriptor`s we stringify. (I agree with avoiding the implicit callback, it was a mistake on my part.)

Comment (arcanis) [reply packages/plugin-essentials/sources/commands/remove.ts:92]: Woops my bad! Wouldn't make sense to stringify a stringified ident indeed 😄

**Comment by paul-soporan** [comment]
Just so you know, I'm in the process of adding a few tests, so try not merging this in the next 20 minutes. 😄

**Comment by paul-soporan** [comment]
Done. 👍

**Comment by arcanis** [comment]
Awesome, thanks for adding tests!

**Pull request merged**